app/helpers/course/assessment/answer/programming_helper.rb

```diff
@@ -37,7 +37,7 @@ def insert_annotations(code_block_html, programming_answer_file)
     document = Nokogiri::XML.parse(code_block_html)
 
     programming_answer_file.annotations.each do |annotation|
-      insert_annotation_block(document, annotation)
+      insert_annotation_block(document, annotation) unless annotation.posts.empty?
```

Comment:
The same could be said for those that are looking at the partial (rather than this helper) and figuring out 'why isn't there a block being displayed'? 

In either case, I'd have a preference to leave this check within the partial.